extensions/cxf/deployment/src/main/java/io/quarkus/cxf/deployment/CxfProcessor.java

```diff
@@ -0,0 +1,1311 @@
+                            reflectiveClass
+                                    .produce(new ReflectiveClassBuildItem(true, true,
+                                            annotation.value(ANNOTATION_VALUE_INTERCEPTORS).asStringArray()));
+                            break;
+                        case "org.apache.cxf.interceptor.InFaultInterceptors":
+                            HashSet<String> inFaultInterceptors = new HashSet<>(
+                                    Arrays.asList(annotation.value(ANNOTATION_VALUE_INTERCEPTORS).asStringArray()));
+                            cxfServletInfo.getInFaultInterceptors().addAll(inFaultInterceptors);
+                            unremovableBeans.produce(new UnremovableBeanBuildItem(
+                                    new UnremovableBeanBuildItem.BeanClassNamesExclusion(inFaultInterceptors)));
+                            reflectiveClass
+                                    .produce(new ReflectiveClassBuildItem(true, true,
+                                            annotation.value(ANNOTATION_VALUE_INTERCEPTORS).asStringArray()));
+                            break;
+                        default:
+                            break;
+                    }
+                }
+                cxfServletInfos.produce(cxfServletInfo);
+            }
+            if (!cxfEndPointConfig.serviceInterface.isPresent() && !cxfEndPointConfig.implementor.isPresent()) {
+                LOGGER.error("either webservice interface (client) or implementation (server) is mandatory");
+            }
+        }
+
+        for (ClassInfo subclass : index.getAllKnownSubclasses(ABSTRACT_FEATURE)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, subclass.name().toString()));
+        }
+        for (ClassInfo subclass : index.getAllKnownSubclasses(ABSTRACT_INTERCEPTOR)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, subclass.name().toString()));
+        }
+        for (ClassInfo subclass : index.getAllKnownImplementors(DATABINDING)) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, subclass.name().toString()));
+        }
+
+    }
+
+    /**
+     * Create Producer bean managing webservice client
+     * <p>
+     * The generated class will look like
+     *
+     * <pre>
+     * public class FruitWebserviceCxfClientProducer extends AbstractCxfClientProducer {
+     * &#64;ApplicationScoped
+     * &#64;Produces
+     * &#64;Default
+     * public FruitWebService createService() {
+     * return (FruitWebService) loadCxfClient ("org.acme.FruitWebService", "http://localhost/fruit",
+     * "http://myServiceNamespace", "FruitWebServiceName", "http://myPortNamespace", "fruitWebServicePortName");
+     * }
+     *
+     */
+    private void generateCxfClientProducer(BuildProducer<GeneratedBeanBuildItem> generatedBean,
+            String cxfClientProducerClassName, String endpointAddress, String sei, String wsdlUrl) {
```

Comment:
Does that mean the `endpointAddress` has to be known at "build time"?

Reply:
@famod crap, ou are absolutly right, I need to change endpointAddress and wsdlUrl to runtime config and pass it to producer thanks to generatedBeanListener... like done on agroal.